src/Symfony/Component/Security/Http/AccessToken/Cas/Cas2Handler.php

```diff
@@ -50,7 +50,16 @@ public function getUserBadgeFrom(string $accessToken): UserBadge
         $xml = new \SimpleXMLElement($response->getContent(), 0, false, $this->prefix, true);
 
         if (isset($xml->authenticationSuccess)) {
-            return new UserBadge((string) $xml->authenticationSuccess->user);
+            $userIdentifier = (string) $xml->authenticationSuccess->user;
+            $attributes = [];
+            if (isset($xml->authenticationSuccess->attributes)) {
+                // Extract all attributes provided by CAS
+                foreach ($xml->authenticationSuccess->attributes->children($this->prefix) as $key => $value) {
+                    $attributes[$key] = (string) $value;
+                }
+            }
+            // Pass the attributes as third parameter of UserBadge
```

Comment:
The code related to this comment is self-explanatory enough to dispense the comment